trace-obfuscation/src/credit_cards.rs

```diff
@@ -0,0 +1,419 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+/// is_card_number checks if b could be a credit card number by checking the digit count and IIN prefix.
+/// If validateLuhn is true, the Luhn checksum is also applied to potential candidates.
+/// Note: This code is based on the code from datadog-agent/pkg/obfuscate/credit_cards.go
+pub fn is_card_number(s: &str, validate_luhn: bool) -> bool {
```

Comment:
A little usability improvement could be made if `AsRef` trait is used for s
https://doc.rust-lang.org/std/convert/trait.AsRef.html#examples

```suggestion
pub fn is_card_number<T: AsRef<str>>(s: T, validate_luhn: bool) -> bool {
     let s = s.as_ref();
```